platforms/core-runtime/logging/src/main/java/org/gradle/internal/deprecation/DeprecationMessageBuilder.java

```diff
@@ -392,23 +387,50 @@ String formatAdvice(List<String> replacements) {
         }
     }
 
-    static final Pattern ID_CLEAN_PATTERN = Pattern.compile("[A-Za-z]+");
+    public static final char DASH = '-';
 
     public static String createDefaultDeprecationId(String... ids) {
         StringBuilder sb = new StringBuilder();
         for (String id : ids) {
-            if(id == null){
+            if (id == null) {
                 continue;
             }
-            Matcher matcher = ID_CLEAN_PATTERN.matcher(id);
-            while (matcher.find()) {
-                sb.append(TextUtil.toLowerCaseLocaleSafe(matcher.group()));
-                sb.append("-");
+            CharSequence cleanId = getToken(id);
+            if (cleanId.length() > 0) {
+                sb.append(cleanId);
+                sb.append(DASH);
             }
         }
-        return removeEnd(sb.toString(), "-");
+        removeTrailingDashes(sb);
+        return sb.toString();
     }
 
+    private static void removeTrailingDashes(StringBuilder sb) {
+        while (sb.length() > 0 && sb.charAt(sb.length() - 1) == DASH) {
+            sb.setLength(sb.length() - 1);
+        }
+    }
+
+    private static CharSequence getToken(String id) {
```

Comment:
This is not a getter method but one that does string manipulation. I think we should have a more descriptive name, with a comment explaining the expected input and output.